🚀  Release v6.5.0-rc.4

ncoden opened the pull request:
# Foundation v6.5.0 Release Candidate 4

This release provide additional bug fixes and various patches to improve the stability of Foundation.

### Note about this pull request
As this version was built to have the best compatibility with `v6.4.4-rc1`, we used a dedicated branch and manually picked safe changes from the development branch `develop`, excluding API evolutions and breaking changes. These imported changes are the `Use pull request #xxx` commits you can see in this pull request. Development will continue on `develop` and will lead to a future release with all changes, including new features and breaking changes.

### Release Checklist
- [x] Bump versions
- [x] Update dist files
- [x] Update SCSS settings file
▾
- [x] Update documentation
▾
- [x] Prepare release for npm
- [x] Prepare release for Composer
- [x] Prepare release for Rails
  * ⚠️  Must be published on https://github.com/zurb/foundation-rails
- [x] ~~Prepare release for Meteor~~ Do not support pre-releases
  * ⚠️  Must be published before the Git release for `.versions`
- [x] Prepare release for NuGet
  * ⚠️  Must be published with https://www.npmjs.com/package/nuget-publish
▾
- [ ] Released!


## Changes from v6.5.0-rc.3

* 🐛  <samp>#11510</samp> - Follow relative and absolute links in SmoothScroll (@ncoden, #11506)
* 🐛  <samp>#11519</samp> - Clean SmoothScroll listeners on destroy (@ncoden)
* 🐛  <samp>#11529</samp> - Allow Interchange filenames to contain brackets (@peterquentin)
* 🐛  <samp>#11530</samp> - Fix various bugs with Tabs/Accordion deep-linking (@ncoden, #11527)
* 🐛  <samp>#11531</samp> - Fix various bugs with "Tap" events in Touch utility (@ncoden, #11423)
* 🚚  <samp>#11508</samp> - Provide CJS, ESM and ES6+ bundles (@ncoden, #11502)
* 📦  <samp>#11512</samp> - Update dependencies (2018-09-22) (@ncoden)
* 📦  <samp>#11528</samp> - Update and fix `foundation-docs` dependency (@ncoden)
* 📖  <samp>#11500</samp> - Improve description of the Prototype page (@chrisdfe)
* 📖  <samp>#11518</samp> - Fix typos in the Javascript documentation (@ncoden)
* 📖  <samp>#11526</samp> - Use correct namespace for DropdownMenu/OffCanvas events (@ncoden, #11523)

## Contributors
* Christopher Ferris ([@chrisdfe](https://github.com/chrisdfe)) [1]
* Nicolas Coden ([@ncoden](https://github.com/ncoden)) [23]
* Peter Quentin ([@peterquentin](https://github.com/peterquentin)) [1]

Special thanks to:
* Daniel Ruf ([@DanielRuf](https://github.com/DanielRuf))
* Kai Falkowski ([@SassNinja](https://github.com/SassNinja))

ncoden committed:
Merge branch 'master-v6.5' into develop-v6.5

ncoden committed:
Use pull request #11508 from ncoden/chore/transpiled-esm-entry-11502 for v6.5.0

d03b2256a chore: provide CJS, ESM and ES2015+ bundles
abcd7c465 docs: add description of bundle formats in the Javascript doc
795047f10 docs: improve title of section about bundle format in Javascript doc
8d94eb7fb docs: add the package.json entry in Javascript documentation

Signed-off-by: Nicolas Coden <nicolas@ncoden.fr>

ncoden committed:
Use pull request #11512 from ncoden/chore/update-dependencies-2018-09-22 for v6.5.0

4bd819f2b chore: update dependencies (2018-09-22)
8ab5dd8b7 chore: update foundation-docs

Signed-off-by: Nicolas Coden <nicolas@ncoden.fr>

ncoden committed:
Use pull request #11510 from ncoden/fix/smooth-scroll-absolute-links-11506 for v6.5.0

140abba37 fix: follow relative and absolute links in SmoothScroll #11506

Signed-off-by: Nicolas Coden <nicolas@ncoden.fr>

chrisdfe committed:
Use pull request #11500 from chrisdfe/patch-1 for v6.5.0

9195b712c Update prototyping-utilities.md
62d7f5ea2 Update prototyping-utilities.md
a99a0b0be Update prototyping-utilities.md

Signed-off-by: Nicolas Coden <nicolas@ncoden.fr>

ncoden committed:
chore: bump version for for v6.5.0-rc.4

ncoden committed:
chore: generate dist files for v6.5.0-rc.4

ncoden committed:
docs: update documentation for v6.5.0-rc.4

ncoden commented:
This should be the last release candidate for Foundation v6.5.0, making it stable and usable in any enviromnent. Others known bugs will be fixed in later path releases.

@SassNinja @IamManchanda 👋 Could I have you review on this ? Thanks!

DanielRuf commented on docs/pages/javascript.md:
```diff
@@ -39,7 +39,7 @@ Know that they all require `foundation.core.js` to be loaded *first*. Some plugi
 ### Import in JavaScript
 
-Foundation is exported as [UMD modules](http://bob.yexley.net/umd-javascript-that-runs-anywhere/). This means that Foundation and its plugins can be imported in any JavaScript environnement.
+By default, Foundation is exported as [UMD modules](http://bob.yexley.net/umd-javascript-that-runs-anywhere/). This means that Foundation and its plugins can be imported and used in any JavaScript environnement.
```
environnement => environment (probably oversaw this one)

DanielRuf commented on docs/pages/javascript.md:
```diff
@@ -63,6 +63,22 @@ var Foundation = require('foundation-sites');
 
+#### Available formats
+
+Foundation is proposed in bundles of various module formats so you can pick the one that match the best your needs. If you don't know these terms yet, take a look at this [10-minute introduction to module formats in JavaScript](https://www.jvandemo.com/a-10-minute-primer-to-javascript-modules-module-formats-module-loaders-and-module-bundlers/). You will find in the `dist/js` folder the following  ...
```
proposed => provided (I guess you mean this?)

DanielRuf replied:
match the best => matches your needs the best

DanielRuf commented on docs/pages/javascript.md:
```diff
@@ -63,6 +63,22 @@ var Foundation = require('foundation-sites');
+Foundation is proposed in bundles of various module formats so you can pick the one that match the best your needs. If you don't know these terms yet, take a look at this [10-minute introduction to module formats in JavaScript](https://www.jvandemo.com/a-10-minute-primer-to-javascript-modules-module-formats-module-loaders-and-module-bundlers/). You will find in the `dist/js` folder the following  ...
+
+* `foundation.js` <span class="label secondary">UMD</span> <span class="label">Default</span><br>
+  Compatible with most environments and tools (AMD, CJS, ESM...). It works almost everywhere by checking the module format of your environments and then using it, which make the bundle a little heavier.
```
which make => which makes

DanielRuf commented on docs/pages/javascript.md:
```diff
@@ -63,6 +63,22 @@ var Foundation = require('foundation-sites');
+  Everything is transpiled to ES5 but the modules. Dedicated to modern bundlers, like Webpack 2+ or Rollup. They will automatically use this bundle and parse the ES6 modules to remove the unused code (see [tree shaking](#tree-shaking) below).
+
+* `foundation.es6.js` <span class="label secondary">ES6</span> (`esnext` in `package.json`)<br>
+  Unlike the others bundles, this bundle is not transpiled. It contains all the Foundation sources in ES6 in a single file. Use it if you want to manually transpile it for your own targets.
```
others bundles => other bundles

DanielRuf commented on gulp/tasks/deploy.js:
```diff
@@ -93,6 +95,23 @@ gulp.task('deploy:dist:files', function() {
+// Generates JS bundles and puts them in the dist/ folder.
+//
+// In addition to the UMD bundle coming from the build task, the following
+// formats are generared: CJS, ESM, ES6.
```
generared => generated

DanielRuf commented on gulp/tasks/deploy.js:
```diff
@@ -93,6 +95,23 @@ gulp.task('deploy:dist:files', function() {
+//
+// In addition to the UMD bundle coming from the build task, the following
+// formats are generared: CJS, ESM, ES6.
+// See "rollup.config.js" for more informations.
```
informations => information (there is no plural form of information, same for data)

DanielRuf commented on js/foundation.smoothScroll.js:
```diff
@@ -48,10 +48,8 @@ class SmoothScroll extends Plugin {
-            if(!$(this).is('a[href^="#"]'))  {
-                return false;
-            }
+            // Follow the link it does not point to an anchor.
```
it => if it

Probably better: if it is not an anchor

ncoden replied:
> Probably better: if it is not an anchor

There is an hyperlink, and what the link points to, an anchor. We are not testing if the hyperlink is an anchor (with `href="#xxx"`) but it points to it.

MDN and W3C calls these elements "Anchor Element" and "fragment id", but everywhere else the "boorkmark" with `href="#xxx"` is called an anchor, and I never heard of "fragment id" before today 😄

DanielRuf commented on rollup.config.js:
```diff
@@ -0,0 +1,60 @@
+module.exports = [
+
+  // UMD
+  // Compatible with most environements and tools (AMD, CJS, ESM...),
```
environements => environments

ncoden replied:
Done :)

ncoden commented:
Ahah. Thank you @DanielRuf for the typo review. I paid attention to avoid mistakes like this and even used a corrector when working on https://github.com/zurb/foundation-sites/pull/11508, but I guess that there is no better corrector than Daniel 😄.

ncoden committed:
Use pull request #11518 from ncoden/docs/typos-in-javascript-doc for v6.5.0

dbefc0fb2 docs: fix various spelling mistakes in Javascript doc and deploy scripts

Signed-off-by: Nicolas Coden <nicolas@ncoden.fr>

ncoden commented:
@DanielRuf Fixed in #11518, and updated the pull request.

DanielRuf commented:
A few places seem to be not updated. If you are unsure which files, I can list them. Let me know when I can review again.

ncoden committed:
Use pull request #11519 from ncoden/fix/clean-smoothscroll-events-on-destroy for v6.5.0

7d717bfb2 refactor: move SmoothScroll to ES6+ and fix linting issues
fd14999f1 refactor: move SmoothScroll click event handling to its own function
e0fbd09c8 fix: clean SmoothScroll events when destroyed
65c26b102 docs: fix typo in SmoothScroll click handler

Signed-off-by: Nicolas Coden <nicolas@ncoden.fr>

ncoden commented:
@DanielRuf Could you review this again please ?

ncoden commented:
@DanielRuf 👋 Could you review this again please?

ncoden commented:
@DanielRuf Will be merged tomorrow, with or without your final review 😉

ncoden committed:
docs: use event namespace instead of Class name for event doc #11523

Document events with the namespace of the triggered event instead of the class name. There is not ideal but this is better than having a documentation that doesn't match the described event.

This fix is compatible with Foundation <= v6.5.0
Closes https://github.com/zurb/foundation-sites/issues/11523

ncoden committed:
Merge pull request #11526 from ncoden/docs/fake-offcanvas-namespace-11523

docs: use correct namespace for DropdownMenu/OffCanvas events (v6.5)

ncoden committed:
fix: fix "tap" event triggering in Touch (v6.5)

The current implementation actually never triggers the "tap" event. This commit refactor its handling to triggers it when the screen is touched without sliding.

Changes:
- Register the `tap` event in jQuery
- Triggers the `tap` event on `mouseend` if no moves were recorded before.

This is a patch of e07bd4e (#11430) for v6.5.

ncoden committed:
fix: keep the original event datas in Touch tap/swipe events (v6.5)

Keep the `touchstart`/`touchmove`/`touchend` datas (like `target`) for the events that are manually triggered: `tap` and `swipe`.

This resolve a bug in "Reveal" when a `touchend` event inside the reveal modal is transformed to a `tap` event on the overlay because the original `target` is lost when triggered the event on the overlay.

This is a patch of f016a9f (#11430) for v6.5.

ncoden committed:
style: remove extranous linebreak in Touch (v6.5)

This is a patch of 2ae641e (#11430) for v6.5.

ncoden committed:
docs: fix various typo in Touch methods and Reveal tests (v6.5)

This is a patch of 6200d17 (#11430) for v6.5.

ncoden committed:
Merge pull request #11531 from ncoden/fix/close-on-tap-11423-for-v6-5-0

fix: fix various bugs with "Tap" events (v6.5)

ncoden committed:
Use pull request #11528 from ncoden/chore/update-foundation-docs for v6.5.0

f2cb1b0ae chore: update "foundation-docs" dependency
38183468c chore: update lockfiles

Signed-off-by: Nicolas Coden <nicolas@ncoden.fr>

peterquentin committed:
Use pull request #11529 from peterquentin/peterquentin-patch-1 for v6.5.0

a3b25c072 Fix parsing rules, allow filename to contain brackets

Signed-off-by: Nicolas Coden <nicolas@ncoden.fr>

ncoden committed:
Use pull request #11530 from ncoden/fix/tabs-accordion-deep-linking-11527 for v6.5.0

33c55db80 fix: fix scroll to deep-linking default anchor on initialization in Tabs/Accordion
8d4a13ead fix: prevent Tabs/Accordion to scroll to others components' anchors
7bccec66f fix: ensure Tab deep-linking to redirect to hash with #
9fbc2eeb5 test: add visual test for Tabs deep-linking

Signed-off-by: Nicolas Coden <nicolas@ncoden.fr>
